io/eolearn/io/geometry_io.py

```diff
@@ -248,14 +258,13 @@ def dataset_crs(self):
     def _load_vector_data(self, bbox):
         """ Loads vector data from geoDB table
         """
+        prepared_bbox = bbox.transform_bounds(self.dataset_crs) if bbox else None
 
         return self.geodb_client.get_collection_by_bbox(
             collection=self.geodb_collection,
             database=self.geodb_db,
-            bbox=bbox.geometry.bounds,
+            bbox=prepared_bbox.geometry.bounds,
```

Comment:
This will raise an error if `prepared_bbox` is `None`. I suggest changing:

```
bbox_bounds = bbox.transform_bounds(self.dataset_crs).geometry.bounds if bbox else None
```
a few lines above and passing `bbox_bounds` here.

Reply:
Good catch.